pkg/build/api/v1/types.go

```diff
@@ -402,6 +402,11 @@ type SourceBuildStrategy struct {
 
 	// ForcePull describes if the builder should pull the images from registry prior to building.
 	ForcePull bool `json:"forcePull,omitempty"`
+
+	// disableImplicitBuild prevents source-to-image from performing a Docker build operation
```

Comment:
@csrwng suggested moving it to an annotation.  I can do that, but i'd like to talk through this a little more before I do..I kinda hate the number of magic annotations we are accumulating and it makes the implementation code uglier.  Our APIs serve two customers:  our users, and ourselves.  Making our code more complex (and less performant) impacts those users too, and per above, there is a(weak) case for why this is useful to an end user anyway.